DEV/org.openl.core/src/org/openl/binding/impl/IdentifierBinder.java

```diff
@@ -28,8 +33,24 @@ public IBoundNode bind(ISyntaxNode node, IBindingContext bindingContext) throws
         // See http://docs.oracle.com/javase/specs/jls/se8/html/jls-6.html#jls-6.4.2 for details
         // Implementation below tries to follow that specification.
 
-        IOpenField field = bindingContext.findVar(ISyntaxConstants.THIS_NAMESPACE, fieldName, strictMatch);
-
+        IOpenField field = null;
+        try {
+            field = bindingContext.findVar(ISyntaxConstants.THIS_NAMESPACE, fieldName, strictMatch);
+        } catch (AmbiguousFieldException ex) {
+            Collection<IOpenField> matchingFields = ex.getMatchingFields();
+            long arraysCount = matchingFields.stream().filter(e -> e.getType().isArray()).count();
+            if (matchingFields.size() - arraysCount == 1) {
+                Optional<IOpenField> f = matchingFields.stream().filter(e -> !e.getType().isArray()).findFirst();
+                if (f.isPresent()) {
+                    bindingContext.addMessage(OpenLMessagesUtils.newWarnMessage(ex.getMessage(), node));
+                    field = f.get();
+                } else {
+                    throw ex;
+                }
+            } else {
+                throw ex;
+            }

```

Comment:
I think it could be better in bindingContext.findVar method because:
* No needs to catch exception for finding alternative field
* warning can be added directly to the context

Reply:
Binding context doesn't have a knowladge about syntax node is used to compile that prevent to build a right message